libcaf_core/caf/async_mail.test.cpp

```diff
@@ -142,108 +142,6 @@ TEST("delegate message") {
       }
     }
   }
-  SECTION("request message") {
```

Comment:
Are these tests no longer relevant?

Reply:
Hm, I've only seen `request message` here and removed it since those are tested in the `event_based_mail` test suite. However, I might've been a bit trigger heavy. The tests are actually about `delegate`. Let me think about where to move them.